File: ocaml/difference-of-squares/difference_of_squares.ml
let square n = n * n

let square_of_sum n =
    (n * (n + 1)) / 2 |> square

let sum_of_squares n =
    Seq.ints 1 |> Seq.take n |> Seq.fold_left (fun a b -> a + square b) 0

let difference_of_squares n =
    square_of_sum n - sum_of_squares n

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number: int) -> int:
    return sum_first(number) ** 2


def sum_of_squares(number: int) -> int:
    return sum_first(number) * (number * 2 + 1) // 3


def difference_of_squares(number: int) -> int:
    return square_of_sum(number) - sum_of_squares(number)


def sum_first(number: int) -> int:
    return number * (number + 1) // 2
